runtime/major_gc.c

```diff
@@ -1355,8 +1355,10 @@ static void cycle_all_domains_callback(caml_domain_state* domain, void* unused,
   /* If the heap is to be verified, do it before the domains continue
      running OCaml code. */
   if (caml_params->verify_heap) {
-    caml_verify_heap(domain);
+    caml_verify_heap_from_stw(domain);
     caml_gc_log("Heap verified");
+    /* TODO: this barrier is not necessary as there is an
```

Comment:
Are you sure that `caml_verify_heap_from_stw` wouldn't race with the GC code below? I'm sceptical since they both access global roots.

Reply:
I removed the TODO and added your explanation in a comment. Thanks!